shared-libs/contacts/src/people.js

```diff
@@ -1,13 +1,13 @@
 const _ = require('lodash');
-const db = require('../db');
-const config = require('../config');
-const utils = require('./utils');
+const db = require('./libs/db');
+const config = require('./libs/config');
+const utils = require('./libs/utils');
 const places = require('./places');
-const lineage = require('@medic/lineage')(Promise, db.medic);
+const lineage = require('./libs/lineage');
 const contactTypeUtils = require('@medic/contact-types-utils');
 
 const getPerson = id => {
-  return lineage.fetchHydratedDoc(id)
+  return lineage.get().fetchHydratedDoc(id)
```

Comment:
The short answer is, no, this is not just because of testing.  

The problem has to do with initializing `@medic/lineage` with the `medic` property from `db` _when the file is loaded_ vs here _at runtime_.  In certain cases, `db.medic` is not initialized when the file is loaded and so directly initializing  `@medic/lineage` will result in errors later when it gets used.  

E.g.: here is the stacktrace from API when trying to create a user:

```
2022-10-04 19:56:12 [31mERROR[39m: Server error: TypeError: Cannot read properties of undefined (reading 'query')
    at fetchLineageById (/shared-libs/lineage/src/hydration.js:237:15)
    at fetchHydratedDoc (/shared-libs/lineage/src/hydration.js:282:12)
    at Object.fetchHydratedDoc (/shared-libs/lineage/src/hydration.js:486:50)
    at Object.getPerson [as _getPerson] (/shared-libs/contacts/src/people.js:14:24)
    at /shared-libs/contacts/src/people.js:102:26
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async createUserEntities (/shared-libs/user-management/src/users.js:579:3)
    at async Object.createUser (/shared-libs/user-management/src/users.js:812:12) {
  [stack]: "TypeError: Cannot read properties of undefined (reading 'query')\n" +
    '    at fetchLineageById (/shared-libs/lineage/src/hydration.js:237:15)\n' +
    '    at fetchHydratedDoc (/shared-libs/lineage/src/hydration.js:282:12)\n' +
    '    at Object.fetchHydratedDoc (/shared-libs/lineage/src/hydration.js:486:50)\n' +
    '    at Object.getPerson [as _getPerson] (/shared-libs/contacts/src/people.js:14:24)\n' +
    '    at /shared-libs/contacts/src/people.js:102:26\n' +
    '    at runMicrotasks (<anonymous>)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n' +
    '    at async createUserEntities (/shared-libs/user-management/src/users.js:579:3)\n' +
    '    at async Object.createUser (/shared-libs/user-management/src/users.js:812:12)',
  [message]: "Cannot read properties of undefined (reading 'query')"
```

Definitely open to other ideas on better ways to fix this problem. I think it is related to being multiple layers deep into shared-libs with each trying their best to maintain the `db` object...  This delayed loading of `lineage` was the best I could come up with that did not seem to require any other massive refactoring.

That being said, I will go ahead and update `./libs/lineage` to lazy-load/cache the lineage value so there should not be any kind of performance impact.

Reply:
Oh, I see. It's because you're actually initializing the db at runtime (the init function), but requiring all files immediately. 

I gave this a try locally and did this: 
In user-management init: 
```
const config = require('./libs/config');
const db = require('./libs/db');

module.exports = (sourceConfig, sourceDb) => {
  config.init(sourceConfig);
  db.init(sourceDb);

  const bulkUploadLog = require('./bulk-upload-log');
  const roles = require('./roles');
  const tokenLogin = require('./token-login');
  const users = require('./users');

  return {
    bulkUploadLog,
    roles,
    tokenLogin,
    users
  };
};
````

and in contacts init:
```
const config = require('./libs/config');
const db = require('./libs/db');

module.exports = (sourceConfig, sourceDb) => {
  config.init(sourceConfig);
  db.init(sourceDb);
  
  const people = require('./people');
  const places = require('./places');
  return {
    people,
    places,
  };
};
````

So I'm delaying the require until after I initialize my helper libraries. 
What do you think?

Reply:
Alternatively, I might also be able to init the lineage library, similar to the db, and keep the requires where they are.

Reply:
Thanks for the explanation!  This makes a lot more sense now, and I like your solution!  Seems better to move the requires and get rid of the `./libs/lineage` file altogether.  The big downside with dropping `./libs/lineage` is that it makes it harder to mock the lineage functions.  After tinkering a bit, I ended up making a small change to the [lineage index.js file](https://github.com/medic/cht-core/blob/7703-offline-user-replace/shared-libs/lineage/src/index.js#L1) so that it is possible to stub the functions.  Let me know if this is a bad idea for some reason!

Reply:
I think the change in the lineage library will be very confusing to anyone reading it. Having `./libs/lineage` seems like the better compromise, even if it's a one-liner.